ament_cmake_export_dependencies/cmake/ament_cmake_export_dependencies-extras.cmake.in

```diff
@@ -20,20 +20,52 @@ if(NOT _exported_dependencies STREQUAL "")
     if(NOT ${_dep}_FOUND)
       find_package("${_dep}" QUIET REQUIRED)
     endif()
-    if(${_dep}_DEFINITIONS)
-      list_append_unique(@PROJECT_NAME@_DEFINITIONS "${${_dep}_DEFINITIONS}")
-    endif()
-    if(${_dep}_INCLUDE_DIRS)
-      list_append_unique(@PROJECT_NAME@_INCLUDE_DIRS "${${_dep}_INCLUDE_DIRS}")
-    endif()
-    if(${_dep}_LIBRARIES)
-      list(APPEND _libraries "${${_dep}_LIBRARIES}")
-    endif()
-    if(${_dep}_LINK_FLAGS)
-      list_append_unique(@PROJECT_NAME@_LINK_FLAGS "${${_dep}_LINK_FLAGS}")
+    # if a package provides modern CMake interface targets use them
+    # exclusively assuming the classic CMake variables only exist for
+    # backward compatibility
+    set(use_modern_cmake FALSE)
+    if(NOT "${${_dep}_TARGETS}" STREQUAL "")
+      foreach(_target ${${_dep}_TARGETS})
+        # only use actual targets
+        # in case a package uses this variable for other content
+        if(TARGET "${_target}")
+          get_target_property(_include_dirs ${_target} INTERFACE_INCLUDE_DIRECTORIES)
+          if(NOT "${_include_dirs}" STREQUAL "_include_dirs-NOTFOUND")
```

Comment:
@dirk-thomas I believe:
```suggestion
          if(_include_dirs)
```
has the exact same effect. Same elsewhere checking for the `-NOTFOUND` suffix.

Reply:
Done in 0cfa4fab79615e95017ccd1ae0d614035f47bc52.